src/legend.ts

```diff
@@ -157,18 +159,19 @@ export function renderGradientLegend(
           ...state.range.map((color, index: number) => {
             const isMinMax = index === 0 || index === state.range.length - 1;
             const step = formatNumber(rangeStep(state.domain, index, state.range.length));
-            const [min, max] = state.domain;
+            const domain = (state.domain && Array.isArray(state.domain)) ? state.domain : [0, 0]
```

Comment:
Not required, but I think that `Array.isArray(state.domain)` already takes care of checking if `state.domain` is truthy. So that could be simplified to:
```
const domain = Array.isArray(state.domain) ? state.domain : [0, 0]
```

Reply:
good catch! I will change it. Thank you